packages/core/src/features/createTheme.js

```diff
@@ -44,7 +41,14 @@ export const createCreateThemeFunction = (
 			if (cssProps.length && !sheet.rules.themed.cache.has(className)) {
 				sheet.rules.themed.cache.add(className)
 
-				const rootPrelude = style === config.theme ? ':host,' : ''
+				const setRootPrelude = () => {
```

Comment:
I would name this `get*` instead since it only gets a value

```suggestion
				const getRootPrelude = () => {
```

you can even move it to the module scope instead since it's mostly static

```
const getRootPrelude = (isShadowDom: boolean) => isShadowDom ? ':host' : ':root';
```